lib/execution_engine2/db/models/models.py

```diff
@@ -312,15 +312,22 @@ class Job(Document):
 
     terminated_code = IntField(validation=valid_termination_code)
     error_code = IntField(validation=valid_errorcode)
-
+    batch_job = BooleanField(default=False)
     scheduler_type = StringField()
     scheduler_id = StringField()
     scheduler_estimator_id = StringField()
     job_input = EmbeddedDocumentField(JobInput, required=True)
     job_output = DynamicField()
     condor_job_ads = DynamicField()
-    child_jobs = ListField()
-    batch_job = BooleanField(default=False)
+    child_jobs = ListField()  # Only parent container should have child jobs
+    # batch_parent_container = BooleanField(default=False) # Only parent container should have this
+
+    # Only present when a job has been retried and on the retry_parent
+    retry_count = IntField(min_value=0)
```

Comment:
I'm now thinking that it might be very handy to have the retry count on the child as well to make ordering the children easy for the FE. On the child the retry count would be immutable while on the parent it'd be mutable, obviously.

Reply:
My assumption is that this wouldn't be a difficult change, but maybe I'm wrong

Reply:
added to adr

Reply:
https://github.com/kbase/execution_engine2/pull/367/files#diff-370ecf4c171a1ae480f063fbe22e0dbd2c9a418ba7f8cf51dce663887c74196fR53